tests/main/driver/test_restart_fortran.py

```diff
@@ -0,0 +1,248 @@
+import os
```

Comment:
I'm going to skip reviewing this file for now.

Reply:
This test file has a lot going on, and I'm not sure what it's testing. I think you started based on the `test_restart_serial` but what it needs to test is very different from what you need to test (it's mostly testing the fact that we write restarts).

Instead I'd suggest writing a unit test that:
- initializes a FortranRestartConfig referencing the data used in the util tests (you can use `os.path.dirname(os.path.abspath(__file__))` to get the directory of this test file itself, and then use relative paths from there)
- uses that initialized config to generate a driver state (use a generated grid)
- compares one variable for each staggering type against data in the restart netCDF file to ensure data got loaded with the right alignment. Probably u/v/temperature are enough to compare.

Include any other assertions or cases you think are important, e.g. whether pe/peln are overwritten correctly.